src/main/kotlin/com/cosmotech/api/metrics/MonitorServiceAspect.kt

```diff
@@ -32,7 +32,9 @@ class MonitorServiceAspect(private var meterRegistry: MeterRegistry) {
           "connectorId")
 
   @Pointcut(
-      "within(@org.springframework.stereotype.Service *)" + " && within(com.cosmotech..*Impl)")
+      "within(@org.springframework.stereotype.Service *) && within(com.cosmotech..*Impl)" +
+          " && !execution(* com.cosmotech.scenariorun.azure.ScenarioRunServiceImpl.onScenarioDeleted(..))" +
```

Comment:
Could you check if only onScenarioDeleted and findWorkspaceById are the methods that cause problems?

Reply:
these methods cause pb in my case, but other methods would cause pb in other scenario